lib/fs.js

```diff
@@ -545,9 +545,23 @@ function read(fd, buffer, offset, length, position, callback) {
 
   validateOffsetLengthRead(offset, length, buffer.byteLength);
 
-  if (!NumberIsSafeInteger(position))
+  if (position == null)
     position = -1;
 
+  if (typeof position === 'number') {
+    validateInteger(position, 'position');
+  } else if (typeof position === 'bigint') {
+    if (!(position >= -(2n ** 63n) && position <= 2n ** 63n - 1n)) {
+      throw new ERR_OUT_OF_RANGE('position',
+                                 `>= ${-(2n ** 63n)} && <= ${2n ** 63n - 1n}`,
+                                 position);
+    }
```

Comment:
Might be worth put this in a function to avoid duplicating the code twice.

Reply:
Updated :)
I was thinking if I should also add a `validateInt64` to `internal/validators.js` (in a different PR) because that might get rid of some repetitions in the validation.

_Originally posted by @RaisinTen in https://github.com/nodejs/node/pull/36190#discussion_r533416779_

Like this?

Reply:
Sure this can be done in a future PR :)